jgrapht-opt/src/main/java/org/jgrapht/opt/graph/sparse/specifics/SparseGraphSpecifics.java

```diff
@@ -0,0 +1,337 @@
+/*
+ * (C) Copyright 2021-2021, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.opt.graph.sparse.specifics;
+
+import java.util.Set;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphType;
+
+/**
+ * Specifics which provide a sparse graph implementation.
+ * 
+ * @author Dimitrios Michail
+ */
+public interface SparseGraphSpecifics
+{
+    /**
+     * Get the number of edges.
+     * 
+     * @return the number of edges
+     */
+    int edgesCount();
```

Comment:
Since this is a brand new interface, would it make sense to use long instead of int with an eye towards GraphIterables?